src/services/sync/strategies/WebUntis/WebUntisSchoolyearSyncer.js

```diff
@@ -0,0 +1,544 @@
+								date: entry.date,
+								startTime: entry.startTime,
+								endTime: entry.endTime,
+								teacher: name,
+								subject: entry.su[0].longname,
+								room: entry.ro[0].longname,
+							});
+						}
+					}
+				}
+
+				// One teacher for now
+				// break;
+			}
+		} else { // Iterate over rooms
+			for (const room of intermediateData.rooms) {
+				const name = `${room.name} (${room.longName}${room.building !== '' ? `, ${room.building}` : ''})`;
+
+				let timetable = await this.getCustomizableTimeTableFor(4, room.id, {
+					startDate: intermediateData.currentSchoolYear.startDate,
+					endDate: intermediateData.currentSchoolYear.endDate,
+					onlyBaseTimetable: true,
+					klasseFields: ['id', 'longname'],
+					subjectFields: ['id', 'longname'],
+					teacherFields: ['id', 'longname'],
+				});
+
+				/* TODO: Check for change */
+				timetable = timetable.filter(entry => entry.ro.length === 1
+					&& entry.ro[0].id === room.id
+					&& entry.kl.length > 0
+					&& entry.te.length === 1
+					&& entry.su.length === 1);
+				/* END TODO: Check for change */
+	
+				for (const entry of timetable) {
+					for (const classEntry of entry.kl) {
+						const klass = intermediateData.classes.find(k => k.id === classEntry.id);
+	
+						if (klass !== undefined) {
+							klass.timetable.push({
+								date: entry.date,
+								startTime: entry.startTime,
+								endTime: entry.endTime,
+								teacher: entry.te[0].longname,
+								subject: entry.su[0].longname,
+								room: name,
+							});
+						}
+					}
+				}
+
+				// One room for now
+				// break;
+			}
+		}
+
+		// Collect times for each class
+		for (let klass of intermediateData.classes) {
+			const times = [];
+			for (const timetableEntry of klass.timetable) {
+				const newEntry = {
+					teacher: timetableEntry.teacher,
+					subject: timetableEntry.subject,
+					weekday: this.getWeekDay(timetableEntry.date),
+					startTime: this.getStartTime(timetableEntry.startTime),
+					duration: this.getDuration(timetableEntry.startTime, timetableEntry.endTime),
+					room: timetableEntry.room,
+					count: 1,
+				};
+				let entryFound = false;
+				for (const givenEntry of times) {
+					if (givenEntry.weekday === newEntry.weekday
+						&& givenEntry.startTime === newEntry.startTime
+						&& givenEntry.duration === newEntry.duration
+						&& givenEntry.room === newEntry.room) {
+						givenEntry.count += 1;
+						entryFound = true;
+					}
+				}
+				if (!entryFound) {
+					times.push(newEntry);
+				}
+			}
+
+			// Assumption: After 2 entries it is recurring
+			const filteredTimes = times.filter(entry => entry.count >= 2);
+
+			for (let timeEntry of filteredTimes) {
+				const newEntry = {
+					teacher: timeEntry.teacher,
+					class: klass.longName, // note uppercase N here!
+					times: [ {
+						weekday: timeEntry.weekday,
+						startTime: timeEntry.startTime,
+						duration: timeEntry.duration,
+						room: timeEntry.room,
+					} ],
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [There should be no space before ']'.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4061937)